stdlib/private/StdlibCollectionUnittest/CheckMutableCollectionType.swift.gyb

```diff
@@ -907,16 +907,20 @@ func checkPartition_${'Predicate' if predicate else 'WhereElementIsComparable'}(
 
   var c = makeWrappedCollectionWithComparableElement(elements)
 %   end
-
+  let pivotElt  = c.first
+  var pivot = c.startIndex
 %   if predicate:
   let closureLifetimeTracker = LifetimeTracked(0)
-  let pivot = c.partition() {
-    (lhs, rhs) in
-    _blackHole(closureLifetimeTracker)
-    return extract(lhs).value < extract(rhs).value
+  if let first = pivotElt {
+    pivot = c.partition(by: {
+      val in
+      _blackHole(closureLifetimeTracker)
+      return !(extract(val).value < extract(first).value) })
   }
 %   else:
-  let pivot = c.partition()
+  if let first = c.first {
+    pivot = c.partition(by: { !($0 < first) })
+  }
```

Comment:
Is there still a `partition()` without a predicate?  If not, just drop this branch of the test.